extensions/wikia/UserProfilePageV3/js/UserProfilePage.js

```diff
@@ -1,94 +1,100 @@
-var UserProfilePage = {
-	// TODO: use $.nirvana.getJSON instead
-	ajaxEntryPoint: '/wikia.php?controller=UserProfilePage&format=json',
-	questions: [],
-	currQuestionIndex: 0,
-	modal: null,
-	newAvatar: false,
-	closingPopup: null,
-	userId: null,
-	wasDataChanged: false,
-	forceRedirect: false,
-	reloadUrl: false,
-	bucky: window.Bucky('UserProfilePage'),
-	bannerNotification: null,
-
-	// reference to modal UI component
-	modalComponent: {},
-
-	init: function () {
-		'use strict';
-
-		var $userIdentityBoxEdit = $('#userIdentityBoxEdit');
-		UserProfilePage.userId = $('#user').val();
-		UserProfilePage.reloadUrl = $('#reloadUrl').val();
-		require(['BannerNotification'], function (BannerNotification) {
+require(
+	['jquery', 'mw', 'wikia.nirvana', 'wikia.window', 'wikia.tracker', 'bucky', 'BannerNotification', 'wikia.ui.factory'],
+	function ($, mw, nirvana, window, tracker, bucky, BannerNotification, uiFactory) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has 8 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)